OptimizelySDK/Notifications/NotificationCenter.cs

```diff
@@ -0,0 +1,197 @@
+
+        public delegate void FeatureAccessCallback(string featureKey, string userId, UserAttributes userAttributes, Variation variation);
+        
+        private ILogger Logger;
+
+        // Notification Id represeting number of notifications.
+        public int NotificationId { get; private set; } = 1;
+
+        // Associative array of notification type to notification id and notification pair
+        private Dictionary<NotificationType, Dictionary<int, object>> Notifications = 
+            new Dictionary<NotificationType, Dictionary<int, object>>();
+
+        public int NotificationsCount
+        {
+            get
+            {
+                int notificationsCount = 0;
+                foreach (var notificationsMap in Notifications.Values)
+                {
+                    notificationsCount += notificationsMap.Count;
+                }
+
+                return notificationsCount;
+            }
+        }
+
+        /// <summary>
+        /// NotificationCenter constructor
+        /// </summary>
+        /// <param name="logger">The logger object</param>
+        public NotificationCenter(ILogger logger = null)
+        {
+            Logger = logger ?? new NoOpLogger();
+
+            foreach (NotificationType notificationType in Enum.GetValues(typeof(NotificationType)))
+            {
+                Notifications[notificationType] = new Dictionary<int, object>();
+            }
+        }
+
+        /// <summary>
+        /// Add a notification callback to the notification center.
+        /// </summary>
+        /// <param name="notificationType">Notification type</param>
+        /// <param name="decisionCallBack">Callback function to call when event gets triggered</param>
+        /// <returns></returns>
+        public int AddNotification(NotificationType notificationType, DecisionCallback decisionCallBack)
+        {
+            return AddNotification(notificationType, (object)decisionCallBack);
+        }
+
+        /// <summary>
+        /// Add a notification callback to the notification center.
+        /// </summary>
+        /// <param name="notificationType">Notification type</param>
+        /// <param name="trackCallback">Callback function to call when event gets triggered</param>
+        /// <returns></returns>
+        public int AddNotification(NotificationType notificationType, TrackCallback trackCallback)
+        {
+            return AddNotification(notificationType, (object)trackCallback);
+        }
+
+        /// <summary>
+        /// Add a notification callback to the notification center.
+        /// </summary>
+        /// <param name="notificationType">Notification type</param>
+        /// <param name="featureAccessCallback">Callback function to call when event gets triggered</param>
+        /// <returns></returns>
+        public int AddNotification(NotificationType notificationType, FeatureAccessCallback featureAccessCallback)
```

Comment:
AddNotification(NotificationType.Track, featureAccessCallback)
Here we are providing NotificationType argument which is expected for 1st arg. 
andn in the 2nd we are providing featureAccessCallback which is also expected. 
The problem is notificationType value doesn't match with the featureAccescallback.

Reply:
Validation added for notification type.